src/renderer/atlas/BackendD2D.cpp

```diff
@@ -158,20 +180,33 @@ void BackendD2D::_handleSettingsUpdate(const RenderingPayload& p)
     _generation = p.s.generation();
     _fontGeneration = p.s->font.generation();
     _cursorGeneration = p.s->cursor.generation();
+    _miscGeneration = p.s->misc.generation();
     _viewportCellCount = p.s->viewportCellCount;
 }
 
 void BackendD2D::_drawBackground(const RenderingPayload& p)
 {
     if (_backgroundBitmapGeneration != p.colorBitmapGenerations[0])
     {
-        THROW_IF_FAILED(_backgroundBitmap->CopyFromMemory(nullptr, p.backgroundBitmap.data(), gsl::narrow_cast<UINT32>(p.colorBitmapRowStride * sizeof(u32))));
+        const D2D1_RECT_U rect{
+            1u,
+            1u,
+            1u + p.s->viewportCellCount.x,
+            1u + p.s->viewportCellCount.y,
+        };
+        THROW_IF_FAILED(_backgroundBitmap->CopyFromMemory(&rect, p.backgroundBitmap.data(), gsl::narrow_cast<UINT32>(p.colorBitmapRowStride * sizeof(u32))));
         _backgroundBitmapGeneration = p.colorBitmapGenerations[0];
     }
 
     // If the terminal was 120x30 cells and 1200x600 pixels large, this would draw the
-    // background by upscaling a 120x30 pixel bitmap to fill the entire render target.
-    const D2D1_RECT_F rect{ 0, 0, static_cast<f32>(p.s->targetSize.x), static_cast<f32>(p.s->targetSize.y) };
+    // background by upscaling a 122x32 pixel bitmap to fill the entire render target,
+    // with some overhang on all four sides to paint the gutters.
+    const D2D1_RECT_F rect{
+        static_cast<f32>(-(p.s->font->cellSize.x)),
+        static_cast<f32>(-(p.s->font->cellSize.y)),
+        static_cast<f32>(p.s->targetSize.x + (2 * p.s->font->cellSize.x)),
+        static_cast<f32>(p.s->targetSize.y + (2 * p.s->font->cellSize.y))

```

Comment:
This isn't quite correct if the `targetSize` is not a multiple of the `cellSize`. I think this should be `(cellCount + 1) * cellSize`.

Reply:
Oh, I thought it was the right size - it is the size of the rect we are already filling before this change.

Is there a case where it was correct to fill targetSize before but it is now incorrect to fill targetSize plus padding?

Reply:
Oh, good point. I now remembered that the texture doesn't get scaled by changing the size. It gets copied over 1:1 and the parts that are outside the bounds get filled with `D2D1_EXTEND_MODE_CLAMP`.

But in that case you can still set the right/bottom to match the `targetSize`. There's no reason why it needs to draw beyond that. The offset in left/top is only necessary to get the right offset in after all.

Reply:
But it's using `D2D1_EXTEND_MODE_MIRROR`, not `CLAMP`.

Doesn't the rectangle need to cover the entire region that we're painting with the bitmap?

Reply:
It only needs to cover whatever you want to fill. If you have a 100x100 bitmap and draw it into a 200x200 target, it'll be drawn in the top-left corner as 100x100 and the rest will be extended as specified. That's why we only need to specify a size that barely covers the swap chain. The reason it's scaled up for us is because of the `_backgroundBrush->SetTransform` call.

Now that I think about it, how does this PR work given that you offset it both in the transform and in the target coordinates? 🤔 I'll have to check it out. 😄

Good point about `MIRROR`. I didn't realize I used that. Now that we have a border we can use `CLAMP` for sure.

Reply:
I was baffled by this as well, actually. I set the rect to -w,-h _before_ realizing that the bitmap transform needed to be changed.

The best explanation I could come up with was that the bitmap brush is relative to the global coordinate system.

Reply:
Noted with a comment. Applied the other fixes!